src/Sylius/Bundle/CoreBundle/DependencyInjection/Compiler/OverrideResourceStateMachinePass.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\DependencyInjection\Compiler;
+
+use Sylius\Bundle\CoreBundle\Resource\StateMachine\CompositeStateMachine;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class OverrideResourceStateMachinePass implements CompilerPassInterface
```

Comment:
I think it's important to specify "resource controller SM" and not just "resource SM" as it's a separate configuration option.

```suggestion
final class OverrideResourceControllerStateMachinePass implements CompilerPassInterface
```
```suggestion
final class SymfonyWorkflowResourceControllerStateMachinePass implements CompilerPassInterface
```
```suggestion
final class SymfonyWorkflowOverrideResourceControllerStateMachinePass implements CompilerPassInterface
```

Reply:
It's `RegisterStateMachinePass` inside the Resource bundle, so I don't think it will change anything, but I can change it for you :P